overlord/hookstate/ctlcmd/services.go

```diff
@@ -57,21 +60,52 @@ func (a byApp) Less(i, j int) bool {
 	return a[i].Name < a[j].Name
 }
 
+var newStatusDecorator = func(ctx context.Context, isGlobal bool, uid string) clientutil.StatusDecorator {
+	if isGlobal {
+		return servicestate.NewStatusDecorator(progress.Null)
+	} else {
+		return servicestate.NewStatusDecoratorForUid(progress.Null, ctx, uid)
```

Comment:
This route is not covered in testing.

Reply:
This function is mocked intentionally - if we use `NewStatusDecoratorForUid` then it will start to contact the snapd user client session which is unfortunate to start spinning up in all tests that use this. Instead the mock verifies that `newStatusDecorator` is invoked with the expected arguments, and fakes the responses from from the status decorator.

The `NewStatusDecoratorForUid` functionality, along with the user-session agent is tested independently.